src/api.rs

```diff
@@ -0,0 +1,41 @@
+use std::net::SocketAddr;
+
+use jsonrpsee::server::{RpcModule, ServerBuilder};
```

Comment:
We don't need to use JSON-RPC for the API of this service. This is a design decision and I'm not sure what the advantage of JSON-RPC is.

This particular library (IIRC) is also quite opinionated and forces you to adopt certain patterns that may not be a good fit. I think generally we want to stick to the Tokio/Hyper/Tower ecosystem. Maybe we can consider Tonic (though gRPC has it's own drawbacks)

Reply:
I don't have strong opinions here, I picked JSON-RPC because it seems to be the rough industry standard, all clients will already have a JSON-RPC library and won't have trouble pointing it at tx-sitter.

I had no particular reason for picking jsonrpsee other than it seemed full-featured and active and an easy choice to revert; very happy to go with a less opinionated library, especially [a simpler one based on Tower/Hper](https://docs.rs/jsonrpc-http-server/latest/jsonrpc_http_server/struct.ServerHandler.html).

gRPC does look like an improvement to JSON-RPC though, I'll look into tonic